bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -842,6 +862,64 @@ func (r *KubeadmConfigReconciler) resolveUsers(ctx context.Context, cfg *bootstr
 	return collected, nil
 }
 
+func (r *KubeadmConfigReconciler) resolveDiscoveryKubeConfig(config *bootstrapv1.KubeadmConfig) (*bootstrapv1.File, error) {
+	cfg := config.Spec.JoinConfiguration.Discovery.File
+	if cfg == nil || cfg.KubeConfig == nil {
+		return nil, errors.New("no discovery configuration file to resolve")
+	}
+	cluster := clientcmdv1.Cluster{
+		Server:                   cfg.KubeConfig.Cluster.Server,
```

Comment:
I know its an error case (that I think shouldn't occur), but this panics if cfg.KubeConfig.Cluster is nil. Let's add it to the if in l.867?

Reply:
added a check